File: VB/E1883/PersistentClasses.vb
Imports DevExpress.Xpo

Namespace DXSample

    Public Class Person
        Inherits XPObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Private nameField As String

        Public Property Name As String
            Get
                Return nameField
            End Get

            Set(ByVal value As String)
                SetPropertyValue(Of String)("Name", nameField, value)
            End Set
        End Property

        Private ageField As Integer

        Public Property Age As Integer
            Get
                Return ageField
            End Get

            Set(ByVal value As Integer)
                SetPropertyValue(Of Integer)("Age", ageField, value)
            End Set
        End Property
    End Class
End Namespace

File: CS/E1883/PersistentClasses.cs
using DevExpress.Xpo;
namespace DXSample {
    public class Person :XPObject {
        public Person(Session session) : base(session) { }

        private string name;
        public string Name {
            get { return name; }
            set { SetPropertyValue<string>("Name", ref name, value); }
        }

        private int age;
        public int Age {
            get { return age; }
            set { SetPropertyValue<int>("Age", ref age, value); }
        }
    }
}